### PR TITLE
Introduce `BlockWeightsForAsyncBacking` and `MAXIMUM_BLOCK_WEIGHT_FOR_ASYNC_BACKING` to bridges primitives

### PR DESCRIPTION
**Pull request by bkontur**
Fixes: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4943591

### CODE REVIEW

File: bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs [hunk 1]
```diff
@@ -54,6 +54,14 @@ const MAXIMUM_BLOCK_WEIGHT: Weight = Weight::from_parts(constants::WEIGHT_REF_TI
 	.saturating_div(2)
 	.set_proof_size(polkadot_primitives::MAX_POV_SIZE as u64);
 
+/// We allow for 2 seconds of compute with a 6 second average block.
+///
+/// This is a copy-paste from the `polkadot-sdk` repo where all system parachains are switched to async backing (https://github.com/paritytech/polkadot-sdk/pull/2949).
```
Comment (mrcnski) [inline bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs:59]: > This is a copy-paste from the `polkadot-sdk` repo

This is confusing. Isn't this PR also on `polkadot-sdk`?

File: bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs [hunk 2]
```diff
@@ -54,6 +54,14 @@ const MAXIMUM_BLOCK_WEIGHT: Weight = Weight::from_parts(constants::WEIGHT_REF_TI
 	.saturating_div(2)
 	.set_proof_size(polkadot_primitives::MAX_POV_SIZE as u64);
 
+/// We allow for 2 seconds of compute with a 6 second average block.
+///
```
Comment (bkontur) [inline bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs:58]:
```suggestion
```

File: bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs (hunk 1 above)
Comment (bkontur) [inline bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs:59]:
```suggestion
/// This is a copy-paste of system parachain runtimes switched to async backing (https://github.com/paritytech/polkadot-sdk/pull/2949).
```

File: bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs [hunk 3]
```diff
@@ -54,6 +54,13 @@ const MAXIMUM_BLOCK_WEIGHT: Weight = Weight::from_parts(constants::WEIGHT_REF_TI
 	.saturating_div(2)
 	.set_proof_size(polkadot_primitives::MAX_POV_SIZE as u64);
 
+/// We allow for 2 seconds of compute with a 6 second average block.
+/// This is a copy-paste of system parachain runtimes switched to async backing (https://github.com/paritytech/polkadot-sdk/pull/2949).
```
Comment (bkontur) [inline bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs:58]:
```suggestion
///
/// This is a copy-paste of system parachain runtimes switched to async backing (https://github.com/paritytech/polkadot-sdk/pull/2949).
```

**Review by svyatonik — APPROVED** [review]

File: bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs (hunk 1 above)
Comment (bkontur) [reply bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs]: Changed

**Review by acatangiu — APPROVED** [review]

File: bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs [hunk 4]
```diff
@@ -54,6 +54,14 @@ const MAXIMUM_BLOCK_WEIGHT: Weight = Weight::from_parts(constants::WEIGHT_REF_TI
 	.saturating_div(2)
 	.set_proof_size(polkadot_primitives::MAX_POV_SIZE as u64);
 
+/// We allow for 2 seconds of compute with a 6 second average block.
+///
+/// This is a copy-paste of system parachain runtimes switched to async backing (https://github.com/paritytech/polkadot-sdk/pull/2949).
+const MAXIMUM_BLOCK_WEIGHT_FOR_ASYNC_BACKING: Weight = Weight::from_parts(
```
Comment (bkchr) [inline bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs:60]: This makes no sense? Either the runtime is running using async backing or not. This means you should update `BlockWeights` directly.

Comment (bkontur) [reply bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs:60]: Runtime's `BlockWeights` are updated directly in that https://github.com/paritytech/polkadot-sdk/pull/2949.

Here we try to keep bridges primitives in sync.
Bridges repo is [partially migrated](https://github.com/paritytech/parity-bridges-common/issues/2538) to the `polkadot-sdk`, the relayer part is missing, and we use these primitives for relayer code also (I will double-check if it is really needed now), that's the reason we have these "duplicite" copies. From time to time, we sync/backport bridges repo with polkadot-sdk.

Comment (bkchr) [reply bridges/primitives/chain-bridge-hub-cumulus/src/lib.rs:60]: Hmm okay.

**Pull request merged**